### PR TITLE
Make the UI less of a hot mess

### PR DESCRIPTION
**Pull request by mjgiarlo**
# Why was this change made?

Quality of life: to make the UI arguably prettier and easier to grok, rather than a mess of newlines and indentation.

## Before

hot_mess

### Dark Mode

dark_mode

### Light Mode

light_mode

# How was this change tested?

Ran many deploys

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,14 +5,17 @@ source 'https://rubygems.org'
...
 
 gem 'byebug'
-gem 'config', '~> 3.1'
-gem 'parallel', '~> 1.21'
-gem 'pastel', '~> 0.8'
-gem 'rubocop', '~> 1.24'
-gem 'thor', '~> 1.1'
-gem 'tty-progressbar', '~> 0.18'
-gem 'tty-prompt', '~> 0.23'
-gem 'tty-table', '~> 0.12'
```
Comment (mjgiarlo) [inline Gemfile:15]: I couldn't see any reason to pin these so tightly in a repo like `sdr-deploy`.

File: Gemfile [hunk 2]
```diff
@@ -5,14 +5,17 @@ source 'https://rubygems.org'
...
+gem 'config'
+gem 'parallel'
+gem 'pastel'
+gem 'rubocop'
+gem 'thor'
+gem 'tty-file'
+gem 'tty-logger'
+gem 'tty-markdown'
```
Comment (mjgiarlo) [inline Gemfile:15]: These are the three new dependencies, all from the `TTY` family of libraries we were already using here.

File: config/settings.yml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 work_dir: tmp/repos
+light_mode: false
```
Comment (mjgiarlo) [inline config/settings.yml:2]: Make it easy for folks to change the colors used depending on their preferred terminal experience.

File: lib/cocina_checker.rb [hunk 4]
```diff
@@ -18,21 +18,21 @@ def initialize(repos:, tag: nil)
...
-        puts "\t\t#{repo}"
-      end
+      report_table << [
+        version,
+        repos.sort.join("\n")
+      ]
+    end
+
+    if unique_values.empty?
+      render_markdown('## No repositories depend on Cocina')
```
Comment (mjgiarlo) [inline lib/cocina_checker.rb:33]: Add a new informational message when deploying no repos that have Cocina in them.

File: lib/deployer.rb [hunk 5]
```diff
@@ -24,53 +24,53 @@ def initialize(environment:, repos:, tag: nil, before_command: nil)
...
 
   # rubocop:disable Metrics/CyclomaticComplexity
   def deploy_all
-    puts "Repositories: #{repos.map(&:name).join(', ')}"
+    render_markdown('***')
+    render_markdown("# Deploying the following repositories (#{tag || 'default branch'})")
+    render_markdown(repos.map { |repo| "* #{repo.name}" }.join("\n"))
+
+    prompt_user_for_branch_confirmation!
+    prompt_user_for_approval_confirmation!
```
Comment (mjgiarlo) [inline lib/deployer.rb:36]: Move these beneath the deploy-related header so it's clear what they pertain to.

File: lib/deployer.rb [hunk 6]
```diff
@@ -24,53 +24,53 @@ def initialize(environment:, repos:, tag: nil, before_command: nil)
...
-        )
+        Result.new(repo.name, env, status, output)
       end
     end.flatten
 
     auditor.report
 
-    build_report_table!(results)
-
-    puts report_table.render(:unicode)
```
Comment (mjgiarlo) [inline lib/deployer.rb:65]: Replace the table-based output for deploys with log-based output (see screenshots in PR description)

File: lib/deployer.rb [hunk 7]
```diff
@@ -24,53 +24,53 @@ def initialize(environment:, repos:, tag: nil, before_command: nil)
...
-
     results
-      .select { |result| result.status.match?('FAILED') }
+      .select { |result| result.output.match?('FAILED') }
```
Comment (mjgiarlo) [inline lib/deployer.rb:66]: We hit this so rarely that we didn't notice it was busted.

File: lib/deployer.rb [hunk 8]
```diff
@@ -157,22 +166,14 @@ def deploy(env)
...
-      text.sub!(/^ask :branch.+$/, "set :branch, '#{tag}'")
+      TTY::File.replace_in_file('config/deploy.rb', /^ask :branch.+$/, "set :branch, '#{tag}'", verbose: false)
     else
       # Forces the `git:create_release` cap task to use the HEAD ref, which allows
       # different repositories to use different default branches.
-      text.sub!('ask :branch', 'set :branch')
+      TTY::File.replace_in_file('config/deploy.rb', /ask :branch/, 'set :branch', verbose: false)
     end
-
-    File.write('config/deploy.rb', text)
```
Comment (mjgiarlo) [inline lib/deployer.rb:171]: Use `tty-file` to handle the file mangling.

File: lib/auditor.rb [hunk 9]
```diff
@@ -27,9 +27,11 @@ def add_to_store(repo, errors)
   end
 
   def report
-    puts "\n\n------- BUNDLE AUDIT SECURITY REPORT -------"
+    render_markdown('***')
+    render_markdown('# Bundle Audit Security Report')
+
     if @project_errors.empty?
-      puts colorize_success('No bundle security vulns found!')
+      puts colorize_success('No bundle security vulnerabilities found!')
```
Comment (aaron-collier) [inline lib/auditor.rb:34]: Thank you.

**Review by aaron-collier — APPROVED** [review]

**Review by jmartin-sul — COMMENTED** [review]
giving this a glance after it got merged, seems like a nice set of improvements, thanks!

**Pull request merged**